### ISSUE TITLE
JDK 15 'maven-bundle-plugin' throws ConcurrentModificationException - Upgrade to 5.1.1

### ISSUE THREAD

**Issue by fergual** [FasterXML/jackson-databind#2806]
Exception building Jackson-databind or any other project that uses maven-bundle-plugin in JDK15 ea:

```
openjdk version "15-ea" 2020-09-15
OpenJDK Runtime Environment (build 15-ea+32-1525)
```

`[ERROR] Failed to execute goal org.apache.felix:maven-bundle-plugin:4.2.1:bundle (default-bundle) on project jackson-databind: Internal error in maven-bundle-plugin: ConcurrentModificationException -> [Help 1]`

```
[INFO] --- maven-bundle-plugin:4.2.1:bundle (default-bundle) @ jackson-databind ---
[ERROR] An internal error occurred
java.util.ConcurrentModificationException
    at java.util.TreeMap.callMappingFunctionWithCheck (TreeMap.java:742)
    at java.util.TreeMap.computeIfAbsent (TreeMap.java:596)
    at aQute.bnd.osgi.Jar.putResource (Jar.java:288)
    at aQute.bnd.osgi.Jar$1.visitFile (Jar.java:202)
    at aQute.bnd.osgi.Jar$1.visitFile (Jar.java:177)
    at java.nio.file.Files.walkFileTree (Files.java:2804)
    at aQute.bnd.osgi.Jar.buildFromDirectory (Jar.java:176)
    at aQute.bnd.osgi.Jar.<init> (Jar.java:119)
    at aQute.bnd.osgi.Jar.<init> (Jar.java:172)
    at org.apache.felix.bundleplugin.BundlePlugin.getOSGiBuilder (BundlePlugin.java:604)
    at org.apache.felix.bundleplugin.BundlePlugin.buildOSGiBundle (BundlePlugin.java:936)
```

Issue was originally found in bnd-maven-plugin, see  https://github.com/bndtools/bnd/issues/3903

Fix was applied to biz.aQute.bndlib that both bnd-maven-plugin and maven-bundle-plugin depends on.

To solve this issue, please upgrade maven-bundle-plugin to 5.1.1

**Comment by cowtowncoder** [comment]
Thank you for reporting this. I will need to see the best way to do this -- it is easy for upcoming 2.12 (fix in oss-parent), but will probably need different override for 2.11.
Will try to get this done soon.

**Comment by cowtowncoder** [comment]
And for reference, the only mild concern I have is that at some point in the past, bundle plug-in version had to be locked to 2.5.3 since 2.5.4 seem to break something wrt Java 6 compatibility. But I think 4.x resolved whatever issue it was, as `jackson-annotations` (which targets Java 6) does not have override any more (`jackson-jr` has but this should be removed).
Builds will all be from JDK 8, only target for 2 repos (annotations, jackson-jr) target JDK 6; this for Jackson 2.x (3.x will require at least Java 8 for runtime too)

**Comment by GedMarc** [comment]
Time to drop JDK 6 support? Github actions have also dropped JDK 6 completely, and I see the guys as org.json are also facing JDK 6 compilation issues and are dropping it as well :)

**Comment by cowtowncoder** [comment]
JDK 6 support is only left for `jackson-annotations` (where it really does not matter a lot either way) and `jackson-core`, so it's not a huge number. If JDK 8 could not compile to JDK 6 I would not even bother.
Put another way, I don't think there is much upside for those 2 specifically.

But `jackson-databind` is bit different and there the question is that of when to raise the minimum to JDK 8; at least theoretically JDK 7 is fine for runtime. Of actual use cases I think only Android concerns me, although desugaring could perhaps make that a moot point.
As of now, I think of keeping existing baselines for 2.12, make that sort of "LTS" of Jackson 2.x, and lift baselines for 2.13. Maybe that should include `jackson-core` too.

Oh, `jackson-jr` is actually JDK 6 compatible too. That baseline should be fine to raise.

And then there's the question of maybe merging 2 of 3 java 8 modules (java 8 date/time is separate even for jackson 3.x due to its size).

**Comment by cowtowncoder** [comment]
Ok, should be resolved now, 5.1.1 of maven-bundle-plugin used for:

* 2.11: added explicit override in `jackson-bom` (for most components) and `jackson-annotations` directly
* 2.12, master (3.0.0-SNAPSHOT) defined in `oss-parent` uber-parent

Please let me know if there are remaining issues.
